Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -871,7 +990,7 @@ public boolean onTouch(View v, MotionEvent ev) {
 	 */
 	public void closeSession() {
 		if (prefHelper_.getSmartSession()) {
-			if (keepAlive_) {
+			if (keepAlive_ && !isBranchAppInstantiated) { //No need to check for keepAlive_ for auto session management
```

Comment:
closeSession() should not return if "isBranchAppInstantiated = true" .isBranchAppInstantiated is true when developer uses the new session management by extending BranchLinkApp class